tools/rosbash/scripts/rosrun

```diff
@@ -49,7 +49,11 @@ esac
 
 if [[ -n $CMAKE_PREFIX_PATH ]]; then
   IFS=$'\n'
-  catkin_package_libexec_dirs=($(catkin_find --without-underlays --libexec --share "$1" 2> /dev/null))
+  catkin_package_libexec_dirs=($(catkin_find --without-underlays --libexec --share "$1"))
```

Comment:
I would expect that any output to `stderr` will now appear on the console? That is not desired since the invocation might succeed but still print some warning to `stderr`.

Reply:
Suppressing shouldn't be the standard behavior since the user should be required to have an environment where no application succeeds with errors. Could you elaborate more on that case where `catkin_find` prints errors but the overall invocation of `rosrun` succeeds?

Reply:
Oh, I just see that my PR also has the suggested alternative behavior.

Reply:
> this should be considered as a bug on the user side.
> I suggest that the suppression of errors and warnings should be removed since it has the potential to cause even more confusion.

Actually it isn't. If the package with the warning is installed as a Debian package the user has no way to address the problem since it is likely not even their own package. That is why this case intentionally discards `stderr` output. If a package the user has control about has a warning like this there are numerous other places where that warning will be visible.

> An alternative option would be to store the error message of `catkin_find` and print it, if (and only if) `rosrun` fails because of an empty `catkin_package_libexec_dirs` variable.

t would certainly be good to show the `stderr` output in case the invocation fails. But as long as it succeeds the `stderr` output should stay suppressed.

> I just see that my PR also has the suggested alternative behavior.

Hm, I just asked that in the beginning:

> I would expect that any output to stderr will now appear on the console?

Just from reading the patch I was assuming `stderr` would not be suppressed and only `stdout` would be captured. Have you tried that specific case?

Reply:
Mea culpa, so how about this? The existing debug call also prints out the error information in that case. Btw, piping stderr and stdout to two different variables in one call is not directly possible, so we need two executions of `catkin_find`.

```suggestion
  IFS=$'\n'
  catkin_package_libexec_dirs=($(catkin_find --without-underlays --libexec --share "$1" 2> /dev/null))
  debug "Looking in catkin libexec dirs: $IFS$(catkin_find --without-underlays --libexec --share "$1" 2>&1)"
  unset IFS
```

Also, outsourcing `catkin_find --without-underlays --libexec --share "$1"` to a function would make it look cleaner.

Reply:
> we need two executions of `catkin_find`.

That sounds reasonable. (While it technically would be a race I hardly expect the underlying information to change between the invocations. Also the second invocation would only be necessary in case of an error so no overhead for normal operations.)

It would be great if you could update the patch accordingly.

> outsourcing `catkin_find --without-underlays --libexec --share "$1"` to a function would make it look cleaner.

I don't see value in a function for a one-line invocation.

Reply:
I've updated the PR accordingly.